server/chat/background/task.py

```
@@ -1976,31 +1980,44 @@ def cleanup_stale_background_chats() -> Dict[str, Any]:
                     return {"cleaned": 0}
                 
-                # Mark sessions as failed, returning IDs actually updated
-                cursor.execute("""
-                    UPDATE chat_sessions 
-                    SET status = 'failed', updated_at = %s 
-                    WHERE status = 'in_progress' AND updated_at < %s
-                    RETURNING id
-                """, (datetime.now(), stale_threshold))
+                # Mark sessions as failed per-user (respecting RLS)
+                actually_failed_ids = set()
+                for session_id, user_id, incident_id in stale_sessions:
```

Ruff (0.15.10)

[warning] Loop control variable `incident_id` not used within loop body

(B007)